flink-runtime/src/main/java/org/apache/flink/runtime/execution/librarycache/FlinkUserCodeClassLoaders.java

```diff
@@ -31,24 +31,23 @@
  */
 public class FlinkUserCodeClassLoaders {
 
-	public static URLClassLoader parentFirst(URL[] urls) {
-		return new ParentFirstClassLoader(urls);
-	}
-
 	public static URLClassLoader parentFirst(URL[] urls, ClassLoader parent) {
 		return new ParentFirstClassLoader(urls, parent);
 	}
 
-	public static URLClassLoader childFirst(URL[] urls, ClassLoader parent) {
-		return new ChildFirstClassLoader(urls, parent);
+	public static URLClassLoader childFirst(
+		URL[] urls,
+		ClassLoader parent,
+		String[] alwaysParentFirstPatterns) {
+		return new ChildFirstClassLoader(urls, parent, alwaysParentFirstPatterns);
 	}
 
 	public static URLClassLoader create(
-		ResolveOrder resolveOrder, URL[] urls, ClassLoader parent) {
+		ResolveOrder resolveOrder, URL[] urls, ClassLoader parent, String[] alwaysParentFirstPatterns) {
```

Comment:
Make this a vararg parameter for nice APIs during testing ?

Reply:
I specifically didn't make it a vararg parameter to ensure that all call-sites consciously have to deal with this parameter. Making it a varargs makes it easy to forget that you can/should set this.

What do you think? I can still easily make it a vararg.

Reply:
No, I see your argument. Keep it.